pkg/volume/glusterfs/glusterfs_uidallocator.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package glusterfs
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/registry/core/service/allocator"
+)
+
+/*
+
+Reference#
+https://github.com/openshift/origin/blob/master/pkg/security/uidallocator/allocator.go
+
+
+*/
+
+// Interface manages the allocation of ports out of a range. Interface
+// should be threadsafe.
+
+type Interface interface {
+	Allocate(Block) error
+	AllocateNext() (Block, error)
+	Release(Block) error
+}
+
+var (
+	ErrFull            = errors.New("range is full")
+	ErrNotInRange      = errors.New("provided UID range is not in the valid range")
+	ErrAllocated       = errors.New("provided UID range is already allocated")
+	ErrMismatchedRange = errors.New("the provided UID range does not match the current UID range")
+)
+
+type Allocator struct {
+	r     *Range
+	alloc allocator.Interface
+}
+
+// Allocator implements Interface and Snapshottable
+var _ Interface = &Allocator{}
+
+// New creates a Allocator over a UID range, calling factory to construct the backing store.
+func New(r *Range, factory allocator.AllocatorFactory) *Allocator {
+	return &Allocator{
+		r:     r,
+		alloc: factory(int(r.Size()), r.String()),
+	}
+}
+
+// NewInMemory creates an in-memory Allocator
+func NewInMemory(r *Range) *Allocator {
+	factory := func(max int, rangeSpec string) allocator.Interface {
+		return allocator.NewContiguousAllocationMap(max, rangeSpec)
+	}
+	return New(r, factory)
+}
+
+// Free returns the count of port left in the range.
+func (r *Allocator) Free() int {
+	return r.alloc.Free()
+}
+
+// Allocate attempts to reserve the provided block. ErrNotInRange or
+// ErrAllocated will be returned if the block is not valid for this range
+// or has already been reserved.  ErrFull will be returned if there
+// are no blocks left.
+func (r *Allocator) Allocate(block Block) error {
+	ok, offset := r.contains(block)
+	if !ok {
+		return ErrNotInRange
+	}
+
+	allocated, err := r.alloc.Allocate(int(offset))
```

Comment:
I don't see how this reserves more than a single item if you don't pass start and end